src/meta/generateMetaInfo.js

```diff
@@ -1,75 +1,87 @@
 import removeLeadingSlash from '../url/removeLeadingSlash'
+import verifyTrailingSlash from '../url/verifyTrailingSlash'
 
 /**
  * Generate meta info.
  *
- * TODO: Add config argument? Might be good for things like the title pattern/separator `${title} — ${siteName}`
- * TODO: Add more comments.
- *
  * hid in meta tags is needed to prevent duplicate properties
  * https://nuxtjs.org/faq/duplicated-meta-tags/
  * also they have to override nuxt.configs manifest infos, which sets an hid
  * therefore the og:description hid has to be called 'hid: "og:description"' and so on
  */
 
-export default function generateMetaInfo (options, post, locale, route) {
-  const websiteUrl = options.global.website_url || 'website.url'
-  const siteName = options.global.site_title || ''
-  const ogSiteName = options.localized.og_site_name ? options.localized.og_site_name : siteName
-  const title = generateMetaTitle(options, post)
-  const description = generateMetaDescription(options, post)
-  const ogTitle = post && post.og_enabled && post.og_title ? post.og_title : title
-  const ogDescription = post && post.og_enabled && post.og_description ? post.og_description : description
-  const twitterHandle = options.site_twitter_handle || false
-  const image = generateMetaImage(options, post)
-  const hreflangs = generateHreflangs(post, websiteUrl)
-  const canonicalUrl = websiteUrl + removeLeadingSlash(removeTrailingSlash(route))
+export default function generateMetaInfo ({
+  siteSettings = {},
+  post = {},
+  path = '',
+  locale = '',
+  debug = false,
+  titlePattern = true,
+  titlePatternSeparator = '|'
+} = {}) {
+
+  // Check if we have all required data
+  if(!siteSettings || !siteSettings.global || !siteSettings.global.meta_global_site_settings) throw new Error('siteSettings and siteSettings.global.meta_global_site_settings are required')
+  if(!post) throw new Error('post is required')
+  if(!path) throw new Error('path is required')
+  if(!locale) throw new Error('locale is required')
+
+  const _siteSettings = {
+    meta_site_name: '',
+    meta_description_default: '',
+    website_url: '',
+    ...siteSettings.global.meta_global_site_settings,
+    ...siteSettings.localized
+  }
+
+  // Site Name: Use site settings
+  const siteName = _siteSettings.meta_site_name
+
+  // Title: Use post meta title or fallback to post title
+  const title = post.meta_title ? post.meta_title : post.title || ''
+
+  // Description: Use post meta description or fallback to site settings
+  const description = post.meta_description ? post.meta_description : _siteSettings.meta_description_default
+
+  //  Canonical: Construct canonical and ensure we don't mess up the slashes
+  const canonicalUrl = `${verifyTrailingSlash(_siteSettings.meta_website_url)}${removeLeadingSlash(removeTrailingSlash(path))}`
 
   const metaInfo = {
-    title: `${title} — ${siteName}`,
+    title: titlePattern ? `${title} ${titlePatternSeparator} ${siteName}` : siteName,
```

Comment:
maybe someone wants to have `title` and `siteName` without a seperator (just a whitespace). would we then just use an empty `titlePatternSeperator`?